providers/Recommendations.provider.tsx

```diff
@@ -0,0 +1,240 @@
+'use client';
```

Comment:
This file is quite big. What do you think of splitting it into a reducer and provider file?